### ISSUE TITLE
Build issue with .gz file in cordova android

### ISSUE THREAD

**Issue by naveenprincep** [angular/angular-cli#2798]
I am using angular Cli v1.0.0beta17 + Cordova to build a hybrid application.
The debug version is working fine. FYI - I use `ng build` + `cordova build` combinations 

But when I take production version using`ng build --prod` and the to build application with `cordova build andoid` I am getting error.
Further research found out that when ng production build is take (in my case) there are 3 **.gz** files.
Android aapt does not support building gz file in asset folder. Here are the errors i am getting

`Error: cmd: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.
- What went wrong:
  Execution failed for task ':processDebugResources'.
  > com.android.ide.common.process.ProcessException: org.gradle.process.internal.E
  xecException: Process 'command 'E:\Android Studio\sdk\build-tools\24.0.2\aapt.ex
  e'' finished with non-zero exit value 1`

`CatBucketConfig]       Unable to add 'E:\xxxxxxxxxxxxxxxx\platforms\android\assets\www\main.69550f811b9602b43919.bundle.js.gz': file already in archive (try '-u'?)`

Note - I tried building without gz file and the app got build. But the problem is without the gz files the will only show white screen.

Any help will be appreciated. Thank you very much

**Comment by filipesilva** [comment]
The `.gz` files are completely optional and shouldn't affect your app if they are not there. If you're getting a white screen, there's probably something else is wrong.

Try doing `ng build --prod`, deleting the `.gz` files from `dist/` and serving it's contents via a `http-server` or something similar. If you can get a repro that I can use in this scenario, I can help debug.

**Comment by jvitor83** [comment]
TL;DR;

1 - `ng new angular-test`
2 - `cordova create cordova-test cordova.test`
3 - Merge both contents to the same directory
4 - Change the `outDir` of `angular-cli.json` to the `www`
5 - `cordova platform add android`
6 - `ng build --prod`
7 - `cordova run android` (here it break)
_Going further_
8 - `cd platforms/android`
9 - `gradlew build --debug` (on windows)
> Look for the "Unable to add '...\platforms\android\build\intermediates\assets\debug\www\main.adc7a9a050b294c2e7a6.bundle.js.gz': file already in archive"

--------------------

0 - Have installed the Android SDK or Android Studio 'https://developer.android.com/studio/index.html#downloads'
1 - Install globally both CLI (angular and cordova) - `npm i -g angular cordova`

2 - Create the directory 'hybrid-test' - `md hybrid-test`
3 - Enter this directory 'hybrid-test' - `cd hybrid-test`

4 - Create new ANGULAR project - `ng new angular-test`
5 - Enter the 'angular-test' directory
6 - Change the `outDir` of `angular-cli.json` to the `www`
7 - Back to the 'hybrid-test' directory - `cd ..`

8 - Create new CORDOVA project - `cordova create cordova-test cordova.test`
9 - Enter the 'cordova-test' directory - `cd cordova-test`
10 - Create the android platform - `cordova platform add android`
11 - Back to the 'hybrid-test' directory - `cd ..`

12 - Create the directory 'angular-cordova-test' - `mkdir angular-cordova-test`
13 - Copy the CONTENTS of both 'angular-test' and 'cordova-test' to the 'angular-cordova-test'
14 - Enter the 'angular-cordova-test' directory - `cd angular-cordova-test`

15 - Build the angular project to prod - `ng build --prod`
16 - Run the cordova project - `cordova run android`

> ERROR HERE

17 - Go to the android platform directory - `cd platforms/android`
18 - Build using Gradle with debug argument - `gradlew build --debug` (on windows)

> Look for the "Unable to add '...\platforms\android\build\intermediates\assets\debug\www\main.adc7a9a050b294c2e7a6.bundle.js.gz': file already in archive"


----------------------------------------

It already has workaround with the [https://gist.github.com/josx/fc76006e6d877b17fefd](https://gist.github.com/josx/fc76006e6d877b17fefd) hook.
[https://forum.ionicframework.com/t/android-build-failed-ionic-cordova-unable-to-add-asset-file-file-already-in-archive/41146/5](https://forum.ionicframework.com/t/android-build-failed-ionic-cordova-unable-to-add-asset-file-file-already-in-archive/41146/5)

> Would be nice if the Angular CLI come with the `outDir` default to `www`. Make it more easily to create and run an angular project in cordova. And could even be explained at docs.

Don't know if it is something the CLI Team should concern, i think it is an `aapt.exe` error (android sdk build-tools).

**Comment by filipesilva** [comment]
I'm sorry but this is not something we're aiming to specifically fix, no. This is something for a user to configure and tool around given it's a specific, alternate usecase.
